src/set-core.js

```diff
@@ -6,6 +6,33 @@ var assign = require("can-util/js/assign/assign");
 var each = require("can-util/js/each/each");
 var makeArray = require("can-util/js/make-array/make-array");
 var isEmptyObject = require("can-util/js/is-empty-object/is-empty-object");
+var getProps = require("can-util/js/get/get");
+
+// concatUnique
+// concat all items in bItems onto aItems that do not already exist in aItems.
+// same-object and ID collisions are both looked at when deciding whether
+// an item matches another.
```

Comment:
why both same-object and ID?  Wouldn't ID cover both cases?

Reply:
Same object covers the case where there is no ID clause.  Ideally I'd like to cover same-object and equal object by all props when ID is absent, but that would be slower and too much work for the time being.